File: vol2/ch16/LeetCode206.cpp
class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        if(head==NULL || head->next==NULL)
            return head;    //递归出口
        ListNode *p=reverseList(head->next);    //反转子链表
        head->next->next=head;
        head->next=NULL;
        return p;
    }
};

File: vol2/ch16/LeetCode206.py
class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        if not head or not head.next:
            return head;    #递归出口
        p=self.reverseList(head.next)   #反转子链表
        head.next.next=head
        head.next=None
        return p
